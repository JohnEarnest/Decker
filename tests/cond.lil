show[if 1 11 end] # clause permutations
show[if 0 11 end]
show[if 1 11 else 22 end]
show[if 0 11 else 22 end]
show[if 1 11 elseif 0 22 end]
show[if 0 11 elseif 1 22 end]
show[if 0 11 elseif 0 22 end]
show[if 1 11 elseif 0 22 else 33 end]
show[if 0 11 elseif 1 22 else 33 end]
show[if 0 11 elseif 0 22 else 33 end]
show[if 1 11 elseif 0 22 elseif 0 33 end]
show[if 0 11 elseif 1 22 elseif 0 33 end]
show[if 0 11 elseif 0 22 elseif 1 33 end]
show[if 0 11 elseif 0 22 elseif 0 33 end]
show[if 1 11 elseif 0 22 elseif 0 33 else 44 end]
show[if 0 11 elseif 1 22 elseif 0 33 else 44 end]
show[if 0 11 elseif 0 22 elseif 1 33 else 44 end]
show[if 0 11 elseif 0 22 elseif 0 33 else 44 end]
show[if 1 end] # empty bodies
show[if 0 else end]
show[if 0 elseif 1 else end]
show[if 0 elseif 0 else end]
show[if 1 11 else if 0 22 else 33 end end] # nesting
show[if 0 11 else if 1 22 else 33 end end]
show[if 0 11 else if 0 22 else 33 end end]
show[if 1 if 1 11 else 22 end else 33 end]
show[if 1 if 0 11 else 22 end else 33 end]
show[if 0 if 1 11 else 22 end else 33 end]

show[if  nil "yes" else "nope" end] # nil is falsey
show[if !nil "yes" else "nope" end]
