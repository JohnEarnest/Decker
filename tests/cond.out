11

11
22
11
22

11
22
33
11
22
33

11
22
33
44




11
22
33
11
22
33
"nope"
"yes"
